### PR TITLE
Fixes bug in Test::Schema when using filter_parameters

### PR DESCRIPTION
**Pull request by leonelgalan**
By default all Rails apps are created with an initializer called _filter_parameter_logging.rb_ and its default contents are:

```ruby
Rails.application.config.filter_parameters += [:password]
```

If this line is executed, as it is on this PR, the class of `request.filtered_parameters` changes from `ActiveSupport::HashWithIndifferentAccess` to `Hash`. This causes `Schema::Test` not to be able to find schemas by controller path and action:

```
     ActiveModelSerializers::Test::Schema::MissingSchema:
       No Schema file at spec/support/schemas//.json
```

Not properly "mocking" the Rails app properly caused the bug to be hidden in this and [leonelgalan/rspec-active_model_serializers](https://github.com/leonelgalan/rspec-active_model_serializers) test suite.

This bug was first reported by me in another in https://github.com/rails-api/active_model_serializers/pull/1947#pullrequestreview-5497390, but at the time I couldn't produce failing tests.

### CODE REVIEW

File: test/active_model_serializers/test/schema_test.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 require 'test_helper'
 
+Rails.application.config.filter_parameters += [:password]
```
Comment (bf4) [inline test/active_model_serializers/test/schema_test.rb]: @leonelgalan so, this is enough to make the test fail? (this should be set in the test app, I think)

Comment (leonelgalan) [inline test/active_model_serializers/test/schema_test.rb]: I moved into a configure block inside _test/support/rails_app.rb_

File: test/support/rails_app.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ module ActiveModelSerializers
       config.active_support.test_order = :random
       config.action_controller.perform_caching = true
       config.action_controller.cache_store = :memory_store
+
+      config.filter_parameters += [:password]
```
Comment (bf4) [inline test/support/rails_app.rb]: do the tests fail with just this change, and not the fix in `lib/active_model_serializers/test/schema.rb` ?

Comment (leonelgalan) [inline test/support/rails_app.rb]: Yes, they will fail with the message above. I'm not sure how to change this setting for a single test to write a failure case.

**Pull request merged**